### PR TITLE
Plugins: Ensure CallResource responses contain valid Content-Type header

### LINKED ISSUE #57239: Loki Labels or Prometheus Metrics Browser produces the Content Type header two times
**What happened**:

When the Loki labels browser or Prometheus metrics browser are loaded, the API request response has the content type header two times like this:
```
Content-Type: application/json
Content-Type: application/json
```

* Example Request URL: http://localhost:58079/api/datasources/21/resources/labels?start=1666151201441000000&end=1666154801441000000
* Involved Grafana API: https://github.com/grafana/grafana/blob/main/docs/sources/developers/http_api/data_source.md#fetch-data-source-resources-by-id

See this from https://httpwg.org/specs/rfc9110.html#field.content-type
> Although Content-Type is defined as a singleton field, it is sometimes incorrectly generated multiple times, resulting in a combined field value that appears to be a list. Recipients often attempt to handle this error by using the last syntactically valid member of the list, leading to potential interoperability and security issues if different implementations have different error handling behaviors

This results in no error in the web browsers I was testing, but some reverse proxy considers this as invalid response.
```
org.glassfish.jersey.message.internal.HeaderValueException: Too many "Content-Type" header values: "[application/json, application/json]"
```

**What you expected to happen**:

No duplicated content type header in the response.

**How to reproduce it (as minimally and precisely as possible)**:

(1) Open Grafana with the web browser and enable the developer tool
(2) Go to Explore and open any Loki or Prometheus data source
(3) In the network tab of developer tool, find the labels request (e.g. "labels?start=1666151197&end=1666154797")
(4) Check its response header. 

Note 1: Chrome-based browser shows the duplicated header well; Firefox only showed one header.
Note 2: You can do this testing using curl. 

**Environment**:
- Grafana version: 9.1.7
- Data source type & version: Loki 2.6.1, Mimir 2.3.1
- OS Grafana is installed on: Oracle Linux 8 (RHEL variant)
- User OS & Browser: Mac OS X Monterey 12.5.1 / Brave 1.44.101 
- Grafana plugins: n/a
- Others: n/a

Screenshot 1 - Loki data source
Screen Shot 2022-10-18 at 9 44 17 PM

Screenshot 2 - Prometheus data source
Screen Shot 2022-10-18 at 9 43 34 PM

Screenshot 3 - curl 
Screen Shot 2022-10-18 at 10 12 05 PM

https://github.com/grafana/grafana/issues/57239#tasklist-block-b65788fd-aa72-46b2-b63c-198982a7e858

### PR DESCRIPTION
**Pull request by xnyo**
Use canonical MIME header keys for comparisons in CallResource response.

This ensures that the headers forced by the HTTP server are overwritten properly, if already provided by the plugin response.

**Which issue(s) does this PR fix?**:

Fixes #57239

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/api/plugin_resource.go [hunk 1]
```diff
@@ -211,6 +217,11 @@ func (hs *HTTPServer) flushStream(stream callResourceClientResponseStream, w htt
 				}
 			}
 
+			// Make sure a content type always is returned in response
+			if !hasContentType && resp.Status != http.StatusNoContent {
```
Comment (wbrowne) [inline pkg/api/plugin_resource.go:221]: A test case for these would be nice!

Comment (wbrowne) [reply pkg/api/plugin_resource.go:221]: In case they don't already exist 😬 Only just saw the logic is moved from before - so can ignore otherwise

Comment (xnyo) [reply pkg/api/plugin_resource.go:221]: Yep there's already a test case for that:

https://github.com/grafana/grafana/blob/41185ee9861b2cb8dcaee2b0516333d0033c6214/pkg/api/plugins_test.go#L398-L420

**Review by wbrowne — APPROVED** [review]
LGTM!

File: pkg/api/plugin_resource.go (hunk 1 above)
Comment (marefr) [reply pkg/api/plugin_resource.go:221]: Maybe a test case when status is 204 would be nice as well?

**Review by marefr — APPROVED** [review]
Great work! LGTM. Added a comment

**Pull request merged**